### PR TITLE
Grant SSH and SFTP access to groups

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/6029

### CODE REVIEW

**Comment by stephdl** [comment]
Sorry @edospadoni I am a bit stuck on the scrollbar and suggestion, I tried to add 

```
.suggestions-override {
  position: initial !important;
}
```

in different way, either inside SSH.vue or App.vue, but it still not work like I would like, when you have several groups, the suggestion widget make a scrollbar in the right.
After the update of the rpm & nethserver-directory, create some random groups and do the CL below

```
config setprop sssd ShellOverrideStatus enabled
config setprop sshd sshd AllowGroups ''
```

File: api/system-openssh/update [hunk 1]
```diff
@@ -20,26 +20,38 @@
...
-    echo $data | jq -r ".props.$field"
-}
-
-# Read JSON from stdin and set the record
-/sbin/e-smith/config setprop sshd PasswordAuthentication "$(_get PasswordAuthentication)" PermitRootLogin "$(_get PermitRootLogin)" TCPPort "$(_get TCPPort)"
+my $AllowGroups = '';
+foreach my $g (keys $data->{props}{AllowGroups} ) {
+    if ($data->{props}{AllowGroups}{$g} eq 'ssh') {
+        $AllowGroups .= "$g,";
+    } elsif ($data->{props}{AllowGroups}{$g} eq 'sftp') {
```
Comment (gsanchietti) [inline api/system-openssh/update]:
```suggestion
    } elsif ($data->{'props'}{'AllowGroups'}{$g} eq 'sftp') {
```

File: api/system-openssh/read [hunk 2]
```diff
@@ -26,20 +26,48 @@
...
-    'configuration' => $db->prepare_json('sshd')
+    'configuration' => $db->prepare_json('sshd'),
+    'groups' => [@{listGroups()}],
+    'ShellOverrideStatus' => ($db->get_prop('sssd','ShellOverrideStatus') 
+            || 'disabled') eq 'enabled' ? JSON::true : JSON::false,
+    'Provider' => $provider
 );
+
+my %allow = ();
+foreach my $g (split(',',$out{configuration}{props}{AllowGroups})) {
```
Comment (gsanchietti) [inline api/system-openssh/read]: Please document the `AllowGroups` prop format inside OpenSSH README

[the remark this answers is not in the document]
Comment (stephdl) [reply api/system-openssh/read]: ok

File: api/system-openssh/read [hunk 3]
```diff
@@ -26,20 +26,48 @@
...
+  if (!$group) {
+    next;
+  }
+  if ($sftp eq 'sftp') {
+    $allow{$group} = 'sftp';
+  } else {
+    $allow{$group} = 'ssh';
+  }
+}
+$out{configuration}{props}{AllowGroups} = \%allow;
```
[the remark this answers is not in the document]
Comment (stephdl) [reply api/system-openssh/read]: ok

File: api/system-openssh/update [hunk 4]
```diff
@@ -20,26 +20,38 @@
...
-function _get
-{
-    field=$1
-    echo $data | jq -r ".props.$field"
-}
-
-# Read JSON from stdin and set the record
-/sbin/e-smith/config setprop sshd PasswordAuthentication "$(_get PasswordAuthentication)" PermitRootLogin "$(_get PermitRootLogin)" TCPPort "$(_get TCPPort)"
+my $AllowGroups = '';
+foreach my $g (keys $data->{props}{AllowGroups} ) {
```
[the remark this answers is not in the document]
Comment (stephdl) [reply api/system-openssh/update]: ok

File: api/system-openssh/update (hunk 1 above)
Comment (stephdl) [reply api/system-openssh/update]: ok

File: api/system-openssh/update [hunk 5]
```diff
@@ -20,26 +20,38 @@
...
+    if ($data->{props}{AllowGroups}{$g} eq 'ssh') {
```
[the remark this answers is not in the document]
Comment (stephdl) [reply api/system-openssh/update]: ok

File: api/system-openssh/update [hunk 6]
```diff
@@ -20,26 +20,38 @@
...
+my $cdb = esmith::ConfigDB->open() || die("Could not open ConfigDB");
+my $action = $data->{'action'};
 
-data=$(cat /dev/stdin)
+foreach my $prop (qw(
+    PasswordAuthentication
+    PermitRootLogin
+    TCPPort
+    )) {
+        $cdb->set_prop('sshd', $prop, $data->{props}{$prop});
```
[the remark this answers is not in the document]
Comment (stephdl) [reply api/system-openssh/update]: ok

File: api/system-openssh/read (hunk 2 above)
Comment (stephdl) [reply api/system-openssh/read]: documented to api

**Comment by edospadoni** [comment]
> Sorry @edospadoni I am a bit stuck on the scrollbar and suggestion, I tried to add
> 
> ```
> .suggestions-override {
>   position: initial !important;
> }
> ```
> 
> in different way, either inside SSH.vue or App.vue, but it still not work like I would like, when you have several groups, the suggestion widget make a scrollbar in the right.
> After the update of the rpm & nethserver-directory, create some random groups and do the CL below
> 
> ```
> config setprop sssd ShellOverrideStatus enabled
> config setprop sshd sshd AllowGroups ''
> ```

Try to add only
```css
.suggestions {
  position: initial !important;
}
```

inside SSH.vue file, in the `<style></style`> tag at the bottom of the file.

**Pull request merged**